src/Raven.Server/Documents/Queries/AST/Query.cs

```diff
@@ -432,6 +448,120 @@ public void Step(Span<double> values)
             }
         }
 
+        private void SegmentOnRollup(Span<StatefulTimestampValue> values)
+        {
+            Debug.Assert(values.Length % 6 == 0);
+            var originalNumOfValues = values.Length / 6;
+            if (_count.Count < originalNumOfValues)
+            {
+                for (int i = _count.Count; i < originalNumOfValues; i++)
+                {
+                    _count.Add(0L);
+                    _values.Add(0d);
+                }
+            }
+
+            for (int i = 0; i < originalNumOfValues; i++)
+            {
+                var index = i * 6;
+                StatefulTimestampValue val;
+                switch (Aggregation)
+                {
+                    case AggregationType.Min:
+                        val = values[index + (int)AggregationType.Min];
```

Comment:
You can replace all of these with a single command before the swtich
```suggestion
                        val = values[index + (int)Aggregation];
```